diff --git a/src/baselib/builtins.py b/src/baselib/builtins.py
index a62636a62..cf0872fb6 100644
--- a/src/baselib/builtins.py
+++ b/src/baselib/builtins.py
@@ -3381,17 +3381,22 @@ def ρσ_real_literal(text: _Str) -> Any:
-_BUILTINS_ARRAYLIKE_TAGS = [
-    "[object Int8Array]",
-    "[object Uint8Array]",
-    "[object Uint8ClampedArray]",
-    "[object Int16Array]",
-    "[object Uint16Array]",
-    "[object Int32Array]",
-    "[object Uint32Array]",
-    "[object Float32Array]",
-    "[object Float64Array]",
-    "[object BigInt64Array]",
-    "[object BigUint64Array]",
-    "[object HTMLCollection]",
-    "[object NodeList]",
-    "[object NamedNodeMap]",
-    "[object TouchList]",
-]
+_BUILTINS_ARRAYLIKE_TAGS = runtime.reflect.construct(
+    runtime.set_class,
+    [
+        [
+            "[object Int8Array]",
+            "[object Uint8Array]",
+            "[object Uint8ClampedArray]",
+            "[object Int16Array]",
+            "[object Uint16Array]",
+            "[object Int32Array]",
+            "[object Uint32Array]",
+            "[object Float32Array]",
+            "[object Float64Array]",
+            "[object BigInt64Array]",
+            "[object BigUint64Array]",
+            "[object HTMLCollection]",
+            "[object NodeList]",
+            "[object NamedNodeMap]",
+            "[object TouchList]",
+        ]
+    ],
+)
@@ -3408 +3413 @@ def ρσ_arraylike(value: Any) -> _Bool:
-    return tag in _BUILTINS_ARRAYLIKE_TAGS
+    return _BUILTINS_ARRAYLIKE_TAGS.has(tag)
